From 2a55dea123deb37db5a119415afa441e54182d75 Mon Sep 17 00:00:00 2001
From: Pawel Dembicki <paweldembicki@gmail.com>
Date: Thu, 16 Mar 2023 19:12:08 +0530
Subject: [PATCH 087/281] OpenWrt:
 820-w1-gpio-fix-problem-with-platfom-data-in-w1-gpio.patch

w1: gpio: fix problem with platfom data in w1-gpio

In devices, where fdt is used, is impossible to apply platform data
without proper fdt node.

This patch allow to use platform data in devices with fdt.

Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>

Change-Id: Icf95b0a5d940455027a56238cbf399323d681dab
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 drivers/w1/masters/w1-gpio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/w1/masters/w1-gpio.c b/drivers/w1/masters/w1-gpio.c
index d4632aace402..dc9eb99bf48d 100644
--- a/drivers/w1/masters/w1-gpio.c
+++ b/drivers/w1/masters/w1-gpio.c
@@ -76,7 +76,7 @@ static int w1_gpio_probe(struct platform_device *pdev)
 	enum gpiod_flags gflags = GPIOD_OUT_LOW_OPEN_DRAIN;
 	int err;
 
-	if (of_have_populated_dt()) {
+	if (of_have_populated_dt() && !dev_get_platdata(&pdev->dev)) {
 		pdata = devm_kzalloc(&pdev->dev, sizeof(*pdata), GFP_KERNEL);
 		if (!pdata)
 			return -ENOMEM;
-- 
2.17.1

